src/code/z_en_item00.c

```diff
@@ -1066,43 +1027,45 @@ void Item_DropCollectibleRandom(PlayState* play, Actor* fromActor, Vec3f* spawnP
     }
 
     if (dropId == ITEM00_FLEXIBLE) {
+        // The drop table indices used here are only for the quantity
         if (gSaveContext.health <= 0x10) { // 1 heart or less
             Actor_Spawn(&play->actorCtx, play, ACTOR_EN_ELF, spawnPos->x, spawnPos->y + 40.0f, spawnPos->z, 0, 0, 0,
                         FAIRY_HEAL_TIMED);
             EffectSsDeadSound_SpawnStationary(play, spawnPos, NA_SE_EV_BUTTERFRY_TO_FAIRY, true,
                                               DEADSOUND_REPEAT_MODE_OFF, 40);
             return;
         } else if (gSaveContext.health <= 0x30) { // 3 hearts or less
-            params = 0xB * 0x10;
-            dropTableIndex = 0x0;
+            params = DROP_TABLE_NUM(11);
+            dropTableIndex = 0; // quantity = 3
             dropId = ITEM00_HEART;
         } else if (gSaveContext.health <= 0x50) { // 5 hearts or less
-            params = 0xA * 0x10;
-            dropTableIndex = 0x0;
+            params = DROP_TABLE_NUM(10);
+            dropTableIndex = 0; // quantity = 1
             dropId = ITEM00_HEART;
         } else if ((gSaveContext.magicLevel != 0) && (gSaveContext.magic == 0)) { // Empty magic meter
-            params = 0xA * 0x10;
-            dropTableIndex = 0x0;
+            params = DROP_TABLE_NUM(10);
+            dropTableIndex = 0; // quantity = 1
             dropId = ITEM00_MAGIC_LARGE;
-        } else if ((gSaveContext.magicLevel != 0) && (gSaveContext.magic <= (gSaveContext.magicLevel >> 1))) {
-            params = 0xA * 0x10;
-            dropTableIndex = 0x0;
+        } else if ((gSaveContext.magicLevel != 0) &&
+                   (gSaveContext.magic <= (gSaveContext.magicLevel >> 1))) { // At most half current magic bar
```

Comment:
Actually this looks like a bug, shout-outs to the horrible magic system, this was probably intended to be
```suggestion
                   (gSaveContext.magic <= (gSaveContext.magicCapacity >> 1))) { // At most half current magic bar
```